### PR TITLE
convert to sdk style

### PR DESCRIPTION
**Pull request by pinzart90**
Begin to convert Dynamo projects to sdk style.

### Release Notes

Convert to sdk style

### CODE REVIEW

File: src/Dynamo.All.sln [hunk 1]
```diff
@@ -17,7 +17,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Engine", "Engine", "{A65338
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ProtoAssociative", "Engine\ProtoAssociative\ProtoAssociative.csproj", "{7318D5E5-9D15-4ABE-8A51-92F58D4F0B85}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ProtoCore", "Engine\ProtoCore\ProtoCore.csproj", "{7A9E0314-966F-4584-BAA3-7339CBB849D1}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ProtoCore", "Engine\ProtoCore\ProtoCore.csproj", "{7A9E0314-966F-4584-BAA3-7339CBB849D1}"
```
Comment (pinzart90) [inline src/Dynamo.All.sln:20]: `9A19103F-16F7-4668-BE54-9A1E7A4F7556` is the GUID for sdk style projects

File: src/Config/CS.props [hunk 2]
```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
+      None
+    </ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
```
Comment (pinzart90) [inline src/Config/CS.props:6]: Architecture mismatch warnings because of msil(anycpu) vs amd64 (x64) 
 maybe we can just have one : anycpu

Comment (mjkkirschner) [reply src/Config/CS.props:6]: In the past we needed x64 for CEF - now that we've gotten rid of it I think you're right.

File: src/Dynamo.All.sln (hunk 1 above)
Comment (mjkkirschner) [reply src/Dynamo.All.sln:20]: so this PR only migrating a few projects to sdk style?

Comment (pinzart90) [reply src/Dynamo.All.sln:20]: Yes...I want to do it a few at a time

File: src/Config/CS.props (hunk 2 above)
Comment (pinzart90) [reply src/Config/CS.props:6]: I will remove x64 when I will move the rest of the projects to sdk style

**Comment by pinzart90** [comment]
passed here https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/5144/

File: src/Config/CS_SDK.props [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    <Compile Remove="obj\**" />
+    <EmbeddedResource Remove="obj\**" />
+    <None Remove="obj\**" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="$(SolutionDir)AssemblySharedInfoGenerator\AssemblySharedInfo.cs">
+      <Link>AssemblySharedInfo.cs</Link>
+    </Compile>
+  </ItemGroup>
+</Project>
```
Comment (aparajit-pratap) [inline src/Config/CS_SDK.props:47]: Is this props file collecting all common project properties in one place?

Comment (pinzart90) [reply src/Config/CS_SDK.props:47]: yes...same concept as CS.props (but for sdk style)

File: src/Engine/GraphLayout/GraphLayout.csproj [hunk 4]
```diff
@@ -1,63 +1,12 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (aparajit-pratap) [inline src/Engine/GraphLayout/GraphLayout.csproj:1]: What's the advantage of using the sdk style project - I see there's multitargeting, any other?

File: src/Engine/ProtoCore/ProtoCore.csproj [hunk 5]
```diff
@@ -1,264 +1,36 @@
...
-    <Compile Include="Lang\InternalAttributes.cs" />
-    <Compile Include="Lang\Replication\ArgumentAtLevel.cs" />
-    <Compile Include="AttributeEntry.cs" />
-    <Compile Include="BuildStatus.cs" />
-    <Compile Include="CodeBlock.cs" />
-    <Compile Include="CodeGen.cs" />
-    <Compile Include="CodeFile.cs" />
-    <Compile Include="CodeGenDS.cs" />
-    <Compile Include="CodePoint.cs" />
-    <Compile Include="Compiler.cs" />
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/ProtoCore.csproj:118]: Where are the included files specified in the project if not here?

File: src/Engine/ProtoCore/ProtoCore.csproj [hunk 6]
```diff
@@ -1,264 +1,36 @@
...
-    <PlatformTarget>x86</PlatformTarget>
-    <CodeAnalysisUseTypeNameInSuppression>true</CodeAnalysisUseTypeNameInSuppression>
-    <CodeAnalysisModuleSuppressionsFile>GlobalSuppressions.cs</CodeAnalysisModuleSuppressionsFile>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-    <CodeAnalysisRuleSetDirectories>;C:\VS2010\Team Tools\Static Analysis Tools\\Rule Sets</CodeAnalysisRuleSetDirectories>
-    <CodeAnalysisIgnoreBuiltInRuleSets>true</CodeAnalysisIgnoreBuiltInRuleSets>
-    <CodeAnalysisRuleDirectories>;C:\VS2010\Team Tools\Static Analysis Tools\FxCop\\Rules</CodeAnalysisRuleDirectories>
-    <CodeAnalysisIgnoreBuiltInRules>true</CodeAnalysisIgnoreBuiltInRules>
-    <Prefer32Bit>false</Prefer32Bit>
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/ProtoCore.csproj:82]: Is it ok to remove all of these properties?

File: src/Engine/GraphLayout/GraphLayout.csproj (hunk 4 above)
Comment (pinzart90) [reply src/Engine/GraphLayout/GraphLayout.csproj:1]: less code...and will be easier to move to newer versions of dotnet

File: src/Engine/ProtoCore/ProtoCore.csproj (hunk 5 above)
Comment (pinzart90) [reply src/Engine/ProtoCore/ProtoCore.csproj:118]: They are automatically included by the sdk style project

File: src/Engine/ProtoCore/ProtoCore.csproj (hunk 6 above)
Comment (pinzart90) [reply src/Engine/ProtoCore/ProtoCore.csproj:82]: CodeAnalysisRuleS are not useful anymore (IMO)
The rest were merged into the CS_SDK.props file

File: src/Engine/GraphLayout/GraphLayout.csproj (hunk 4 above)
Comment (pinzart90) [reply src/Engine/GraphLayout/GraphLayout.csproj:1]: And I do plan to use multitargeting to have a dotnet6 build as well (at least for the non UI projects)

File: src/Engine/ProtoCore/ProtoCore.csproj [hunk 7]
```diff
@@ -1,264 +1,36 @@
...
-      <Name>DesignScriptBuiltin</Name>
-      <Private>True</Private>
-    </ProjectReference>
-    <ProjectReference Include="..\..\NodeServices\DynamoServices.csproj">
-      <Project>{ef879a10-041d-4c68-83e7-3192685f1bae}</Project>
-      <Name>DynamoServices</Name>
-      <Private>False</Private>
-    </ProjectReference>
+    <Compile Remove="LexUtils.cs" />
+    <Compile Remove="Parser/AssociativeGraph.cs" />
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/ProtoCore.csproj:28]: What does this do? Where is lexutils.cs coming from?

File: src/Engine/GraphLayout/GraphLayout.csproj [hunk 8]
```diff
@@ -1,63 +1,12 @@
...
-    <Prefer32Bit>false</Prefer32Bit>
-  </PropertyGroup>
-  <ItemGroup>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Xml.Linq" />
-    <Reference Include="System.Data.DataSetExtensions" />
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Data" />
-    <Reference Include="System.Xml" />
```
[the remark this answers is not in the document]
Comment (pinzart90) [reply src/Engine/GraphLayout/GraphLayout.csproj:46]: Yes

File: src/Engine/ProtoCore/ProtoCore.csproj (hunk 7 above)
Comment (pinzart90) [reply src/Engine/ProtoCore/ProtoCore.csproj:28]: These were not compiled in the old format. I decided to keep them around...and not permanently delete them for now.
Not sure if they are useful to keep around though...

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**